internal/brokers/broker.go

```diff
@@ -305,3 +287,76 @@ func (b Broker) validateUILayout(sessionID string, layout map[string]string) (r
 func (b Broker) parseSessionID(sessionID string) string {
 	return strings.TrimPrefix(sessionID, fmt.Sprintf("%s-", b.ID))
 }
+
+// validateAndMarshalUser checks if the specified rawMsg is a valid userinfo and marshals it as a JSON string.
+func (b Broker) validateAndMarshalUser(rawMsg json.RawMessage) (data string, err error) {
```

Comment:
As this also generates the UID and GUID from UUID and UGID, I feel it’s more than just validate/marshal. Need a better name basically :)

Also, I have a weird feeling this really belongs to a broker. I would pass the name as a parameter for now and revisit if we bind it more to the broker, wydt?

Reply:
I'm open to naming suggestions... :joy:

Reply:
Why not `userInfoJsonFromBrokerData(rawMsg json.RawMessage, brokerName string) (userInfo string, err error)`? A little bit verbose, but…